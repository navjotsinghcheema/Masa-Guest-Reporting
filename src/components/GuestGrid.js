import React from 'react';

import Guest from './Guest';

class GuestGrid extends React.Component {
    constructor(props) {
        super(props);        
    }    
    render() {
        return (
            <div className="row">
                <div className="col-lg-4 col-md-4 col-sm-6 col-xs-12">
                    {/* To Add jitter to the containing div */}
                    <div className="col-lg-12 col-md-12 col-sm-12 col-xs-12">
                        {
                            Array.isArray(this.props.guests) ? 
                                this.props.guests.map((guest, index) => {
                                    <Guest key={index} data={guest} blueprint={this.props.metadata} />})
                            : ''
                        }
                    </div>
                </div>
            </div>
        )
    }
}

export default GuestGrid;