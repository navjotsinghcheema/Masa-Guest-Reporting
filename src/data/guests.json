{
    "meta-data": {
      "last_update": "2018-11-09",
      "row_count": 25,
      "payload": [
        {
          "id": "id",
          "label": "Id",
          "type": "number"
        },
        {
          "id": "first_name",
          "label": "First name",
          "type": "string"
        },
        {
          "id": "last_name",
          "label": "Last name",
          "type": "string"
        },
        {
          "id": "email",
          "label": "Email",
          "type": "string"
        },
        {
          "id": "city",
          "label": "City",
          "type": "string"
        },
        {
          "id": "visit_count",
          "label": "Visit count",
          "type": "number"
        },
        {
          "id": "total_spend",
          "label": "Total spend",
          "type": "number"
        },
        {
          "id": "allow_marketing",
          "label": "Allow marketing",
          "type": "boolean"
        },
        {
          "id": "tags",
          "label": "Tags",
          "type": "array"
        }
      ]
    },
    "data": [
      {
        "id": 1,
        "first_name": "Byrom",
        "last_name": "Leggen",
        "email": null,
        "city": "Chat Trakan",
        "visit_count": 7,
        "total_spend": 137.71,
        "allow_marketing": true,
        "tags": ["protocol", "migration", "executive", "Down-sized", "Proactive"]
      },
      {
        "id": 2,
        "first_name": "Lilias",
        "last_name": "Pace",
        "email": "lpace1@toplist.cz",
        "city": "Krasnyye Chetai",
        "visit_count": 22,
        "total_spend": 138.03,
        "allow_marketing": true,
        "tags": ["protocol", "Compatible", "Cloned", "Cloned", "frame"]
      },
      {
        "id": 3,
        "first_name": "Nappy",
        "last_name": "Prate",
        "email": "nprate2@g.co",
        "city": "Funza",
        "visit_count": 14,
        "total_spend": 153.27,
        "allow_marketing": false,
        "tags": [
          "client-driven",
          "encompassing",
          "Vision-oriented",
          "6th generation",
          "Down-sized"
        ]
      },
      {
        "id": 4,
        "first_name": "Elka",
        "last_name": "Vials",
        "email": "evials3@mtv.com",
        "city": "Karangbenda",
        "visit_count": 16,
        "total_spend": 196.73,
        "allow_marketing": false,
        "tags": null
      },
      {
        "id": 5,
        "first_name": "Malva",
        "last_name": "Denekamp",
        "email": "mdenekamp4@tinyurl.com",
        "city": "Tilburg",
        "visit_count": 9,
        "total_spend": 155.93,
        "allow_marketing": true,
        "tags": null
      },
      {
        "id": 6,
        "first_name": "Ardenia",
        "last_name": null,
        "email": "abaford5@cloudflare.com",
        "city": null,
        "visit_count": 14,
        "total_spend": 41.37,
        "allow_marketing": true,
        "tags": ["Streamlined", "function", "algorithm", "frame", "support"]
      },
      {
        "id": 7,
        "first_name": "Wileen",
        "last_name": "Naughton",
        "email": "wnaughton6@blogtalkradio.com",
        "city": "Cedar Rapids",
        "visit_count": 4,
        "total_spend": 122.07,
        "allow_marketing": true,
        "tags": [
          "Realigned",
          "mission-critical",
          "De-engineered",
          "optimal",
          "Polarised"
        ]
      },
      {
        "id": 8,
        "first_name": "Vergil",
        "last_name": "Arnaez",
        "email": "varnaez7@businessweek.com",
        "city": "Kramarzówka",
        "visit_count": 9,
        "total_spend": 115.93,
        "allow_marketing": true,
        "tags": [
          "attitude-oriented",
          "help-desk",
          "neural-net",
          "fault-tolerant",
          "superstructure"
        ]
      },
      {
        "id": 9,
        "first_name": "Mildrid",
        "last_name": "Rubke",
        "email": "mrubke8@ebay.co.uk",
        "city": "Shuangzhong",
        "visit_count": 6,
        "total_spend": 24.96,
        "allow_marketing": true,
        "tags": [
          "Cloned",
          "functionalities",
          "client-server",
          "encompassing",
          "framework"
        ]
      },
      {
        "id": 10,
        "first_name": "Zitella",
        "last_name": "Alfonso",
        "email": "zalfonso9@paginegialle.it",
        "city": "Podsineye",
        "visit_count": 24,
        "total_spend": 162.43,
        "allow_marketing": false,
        "tags": null
      },
      {
        "id": 11,
        "first_name": "Orella",
        "last_name": "Richarson",
        "email": "oricharsona@google.com.hk",
        "city": "Whitehorse",
        "visit_count": 16,
        "total_spend": 183.33,
        "allow_marketing": false,
        "tags": null
      },
      {
        "id": 12,
        "first_name": "Hubey",
        "last_name": "Grossman",
        "email": null,
        "city": "Huaizhong",
        "visit_count": 8,
        "total_spend": 120.75,
        "allow_marketing": false,
        "tags": null
      },
      {
        "id": 13,
        "first_name": "Claudina",
        "last_name": null,
        "email": "cdunhillc@friendfeed.com",
        "city": null,
        "visit_count": 15,
        "total_spend": 132.26,
        "allow_marketing": true,
        "tags": [
          "Visionary",
          "Open-source",
          "functionalities",
          "intermediate",
          "well-modulated"
        ]
      },
      {
        "id": 14,
        "first_name": "Claudine",
        "last_name": "Osgardby",
        "email": "cosgardbyd@smh.com.au",
        "city": "Xiashe",
        "visit_count": 20,
        "total_spend": 73.16,
        "allow_marketing": false,
        "tags": [
          "Re-contextualized",
          "database",
          "Pre-emptive",
          "projection",
          "Robust"
        ]
      },
      {
        "id": 15,
        "first_name": "Nefen",
        "last_name": "Girtin",
        "email": "ngirtine@indiegogo.com",
        "city": "Pom Prap Sattru Phai",
        "visit_count": 15,
        "total_spend": 37.87,
        "allow_marketing": false,
        "tags": ["extranet", "Innovative", "software", "Optimized", "Front-line"]
      },
      {
        "id": 16,
        "first_name": "Bronnie",
        "last_name": "Bargery",
        "email": "bbargeryf@dailymail.co.uk",
        "city": "Göteborg",
        "visit_count": 18,
        "total_spend": 123.77,
        "allow_marketing": true,
        "tags": [
          "Streamlined",
          "motivating",
          "background",
          "radical",
          "Configurable"
        ]
      },
      {
        "id": 17,
        "first_name": "Ronica",
        "last_name": "O'Howbane",
        "email": "rohowbaneg@bloglovin.com",
        "city": "Zeguo",
        "visit_count": 3,
        "total_spend": 153.81,
        "allow_marketing": false,
        "tags": null
      },
      {
        "id": 18,
        "first_name": "Goran",
        "last_name": "Sorley",
        "email": "gsorleyh@jiathis.com",
        "city": "Wukari",
        "visit_count": 14,
        "total_spend": 54.27,
        "allow_marketing": false,
        "tags": ["Phased", "real-time", "Total", "concept", "local"]
      },
      {
        "id": 19,
        "first_name": "Caleb",
        "last_name": null,
        "email": "cwestreyi@shop-pro.jp",
        "city": null,
        "visit_count": 18,
        "total_spend": 52.45,
        "allow_marketing": false,
        "tags": [
          "human-resource",
          "Optional",
          "approach",
          "knowledge user",
          "Graphic Interface"
        ]
      },
      {
        "id": 20,
        "first_name": "Herschel",
        "last_name": null,
        "email": "hcanepej@google.com.hk",
        "city": null,
        "visit_count": 1,
        "total_spend": 98.06,
        "allow_marketing": false,
        "tags": [
          "Reverse-engineered",
          "foreground",
          "Proactive",
          "algorithm",
          "Customer-focused"
        ]
      },
      {
        "id": 21,
        "first_name": "Price",
        "last_name": "Breissan",
        "email": "pbreissank@mashable.com",
        "city": "Maumere",
        "visit_count": 1,
        "total_spend": 136.83,
        "allow_marketing": false,
        "tags": ["Cloned", "hybrid", "intranet", "hierarchy", "policy"]
      },
      {
        "id": 22,
        "first_name": "Alistair",
        "last_name": null,
        "email": "aminerl@foxnews.com",
        "city": null,
        "visit_count": 16,
        "total_spend": 11.5,
        "allow_marketing": false,
        "tags": ["regional", "protocol", "bifurcated", "Automated", "core"]
      },
      {
        "id": 23,
        "first_name": "Silvester",
        "last_name": "Poacher",
        "email": "spoacherm@va.gov",
        "city": "Babat",
        "visit_count": 16,
        "total_spend": 49.0,
        "allow_marketing": false,
        "tags": [
          "Triple-buffered",
          "throughput",
          "Adaptive",
          "focus group",
          "time-frame"
        ]
      },
      {
        "id": 24,
        "first_name": "Emmerich",
        "last_name": "Cure",
        "email": "ecuren@squidoo.com",
        "city": "Ivano-Frankove",
        "visit_count": 16,
        "total_spend": 114.04,
        "allow_marketing": true,
        "tags": [
          "Customer-focused",
          "actuating",
          "Operative",
          "Managed",
          "bifurcated"
        ]
      },
      {
        "id": 25,
        "first_name": "Grady",
        "last_name": "Pauletti",
        "email": "gpaulettio@merriam-webster.com",
        "city": "Xiaoyue",
        "visit_count": 11,
        "total_spend": 170.15,
        "allow_marketing": true,
        "tags": [
          "logistical",
          "object-oriented",
          "holistic",
          "Universal",
          "collaboration"
        ]
      }
    ]
  }  