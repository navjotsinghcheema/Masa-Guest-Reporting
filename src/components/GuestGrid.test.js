import React from 'react';
import renderer from 'react-test-renderer';
import GuestGrid from './GuestGrid';
import guests from '../data/guests.json';

test('Guest snapshot test', () => {
  const metadata = guests["meta-data"];
  const data = guests.data;
  const component = renderer.create(<GuestGrid blueprint={metadata} data={data} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});