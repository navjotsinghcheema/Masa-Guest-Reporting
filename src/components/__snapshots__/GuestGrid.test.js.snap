// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guest snapshot test 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-4 col-md-4 col-sm-6 col-xs-12"
  >
    <div
      className="col-lg-12 col-md-12 col-sm-12 col-xs-12"
    >
      
    </div>
  </div>
</div>
`;
